src/screens/ChannelListScreen/index.tsx

```diff
@@ -1,19 +1,37 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 /* eslint-disable no-use-before-define */
 import * as React from 'react';
-import {FlatList} from 'react-native';
+import {FlatList, StatusBar, View} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
 
+import ChannelListHeaderItem from '../../components/ChatList/ChannelListHeaderItem';
 import CommunityChannelItem from '../../components/ChatList/CommunityChannelItem';
 import GroupChatChannelItem from '../../components/ChatList/GroupChatChannelItem';
 import MessageChannelItem from '../../components/AnonymousChat/MessageChannelItem';
 import PostNotificationChannelItem from '../../components/AnonymousChat/PostNotificationChannelItem';
+import Search from './elements/Search';
 import useFollowUser from './hooks/useFollowUser';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import useRootChannelListHook from '../../hooks/screen/useRootChannelListHook';
 import useSignedChannelListScreenHook from '../../hooks/screen/useSignedChannelListHook';
+import useUserAuthHook from '../../hooks/core/auth/useUserAuthHook';
 
-const ChannelListScreen = () => {
-  const {channels, goToChatScreen, goToPostDetailScreen, goToCommunityScreen} =
-    useSignedChannelListScreenHook();
+const ChannelListScreen = ({route}) => {
+  const {refresh} = useLocalDatabaseHook();
+  const {checkNotificationPermission} = useRootChannelListHook();
   const {handleFollow, isInitialFollowing} = useFollowUser();
+  const {profile} = useUserAuthHook();
+  const navigation = useNavigation();
+  const isFocused = navigation.isFocused();
+  const {channels, goToChatScreen, goToPostDetailScreen, goToCommunityScreen, goToContactScreen} =
+    useSignedChannelListScreenHook();
+
+  React.useEffect(() => {
+    if (isFocused) {
+      refresh('channelList');
+      checkNotificationPermission();
+    }
+  }, [isFocused]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces several hooks and dependencies that were not present in the old code. While this is not inherently problematic, it does increase the complexity of the component. It's important to ensure that these hooks are necessary and that their usage follows best practices for React hooks (e.g., they should not be called conditionally or within loops). 

Also, the `isFocused` variable is used to trigger a refresh and check notification permissions when the screen is focused. This could potentially lead to performance issues if these operations are expensive and the screen gains focus frequently. Consider optimizing these operations if possible.

```diff
const ChannelListScreen = ({route}) => {
  const {refresh} = useLocalDatabaseHook();
  const {checkNotificationPermission} = useRootChannelListHook();
   const {handleFollow, isInitialFollowing} = useFollowUser();
  const {profile} = useUserAuthHook();
  const navigation = useNavigation();
  const isFocused = navigation.isFocused();
  const {channels, goToChatScreen, goToPostDetailScreen, goToCommunityScreen, goToContactScreen} =
    useSignedChannelListScreenHook();

  React.useEffect(() => {
    if (isFocused) {
      refresh('channelList');
      checkNotificationPermission();
    }
  }, [isFocused]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->